webapp/src/components/messages/cloudMessage.tsx

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react'
+
+import {useIntl, FormattedMessage} from 'react-intl'
+
+import {Utils} from '../../utils'
+import IconButton from '../../widgets/buttons/iconButton'
+import Button from '../../widgets/buttons/button'
+
+import CloseIcon from '../../widgets/icons/close'
+
+import {useAppSelector, useAppDispatch} from '../../store/hooks'
+import octoClient from '../../octoClient'
+import {IUser, UserConfigPatch} from '../../user'
+import {getMe, patchProps, getCloudMessageCanceled} from '../../store/users'
+
+import CompassIcon from '../../widgets/icons/compassIcon'
+import TelemetryClient, {TelemetryCategory, TelemetryActions} from '../../telemetry/telemetryClient'
+
+import './cloudMessage.scss'
+const signupURL = 'http://mattermost.com/pricing'
+const displayAfter = (1000 * 60 * 60 * 24) //24 hours
+
+const CloudMessage = React.memo(() => {
+    const intl = useIntl()
+    const dispatch = useAppDispatch()
+    const me = useAppSelector<IUser|null>(getMe)
+    const cloudMessageCanceled = useAppSelector(getCloudMessageCanceled)
+
+    const closeDialogText = intl.formatMessage({
+        id: 'Dialog.closeDialog',
+        defaultMessage: 'Close dialog',
+    })
+
+    const onClose = async () => {
+        if (me) {
+            const patch: UserConfigPatch = {
+                updatedFields: {
+                    focalboard_cloudMessageCanceled: 'true',
+                },
+            }
+
+            const patchedProps = await octoClient.patchUserConfig(me.id, patch)
+            if (patchedProps) {
+                dispatch(patchProps(patchedProps))
+            }
+        }
+    }
+
+    if (me) {
+        const installTime = Date.now() - me.create_at
+        if (Utils.isFocalboardPlugin() || cloudMessageCanceled || installTime < displayAfter) {
```

Comment:
question: for the logic in the nested `if`, wouldn't you want to check `Utils.isFocalboardPlugin() || cloudMessageCanceled` irrespective of `if (me)`?

perhaps something like:


```ts
if (Utils.isFocalboardPlugin() || cloudMessageCanceled) {
    return null
}

if (me) {
    const installTime = Date.now() - me.create_at
    if (installTime < displayAfter) {
        return null
    }
}
```

Reply:
I debated...it really doesn't matter as me should never be undefined, but I do think it makes it more readable and understandable, so I updated.